### ISSUE TITLE
Filename "" does not match expression [A-Za-z\/\.-_]+

### ISSUE THREAD

**Issue by JakeHagen** [scipipe/scipipe#78]
Hello I am trying to use the streamToSubstream functionality but I am getting the error below.
`ERROR   2019/06/12 18:09:20 Filename "" does not match expression [A-Za-z\/\.-_]+`

I also get this error when I try to run your example workflows `https://github.com/pharmbio/scipipe-demo/tree/fdb98884edb98a693c2892930c088cd723070691/dnacanceranalysis`

I believe the issue is in 
```// Run runs the StreamToSubStream
func (p *StreamToSubStream) Run() {
	defer p.CloseAllOutPorts()

	scipipe.Debug.Println("Creating new information packet for the substream...")
	subStreamIP := scipipe.NewFileIP("")
	scipipe.Debug.Printf("Setting in-port of process %s to IP substream field\n", p.Name())
	subStreamIP.SubStream = p.In()

	scipipe.Debug.Printf("Sending sub-stream IP in process %s...\n", p.Name())
	p.OutSubStream().Send(subStreamIP)
	scipipe.Debug.Printf("Done sending sub-stream IP in process %s.\n", p.Name())
}```

Where `subStreamIP := scipipe.NewFileIP("")` is called.
This triggers the error due the `checkFilename` in `NewFileIP` in ip.go

Anything I am doing wrong? thanks for any help you can provide.

**Comment by samuell** [comment]
Thanks for reporting @JakeHagen ! Off arm, it sounds like a bug indeed. Will have to take a closer look shortly though. Stay tuned.

**Comment by JakeHagen** [comment]
Thanks Samuell

**Comment by rbisewski** [comment]
@samuell Confirmed this morning that the [DNA Cancer Analysis demo](https://github.com/pharmbio/scipipe-demo/tree/master/dnacanceranalysis) is broken, and really any bit of code that attempts to open a file or path without a name, such as the [code mentioned by JakeHagen](https://github.com/scipipe/scipipe/blob/master/components/streamtosubstream.go#L36).

Digging into this, the demo appears to be roughly 2 years old, so that didn't change; perhaps there is new code added to the scipipe repo that may have caused a regression?

If I think about this, maybe the commit that broke this was [file checking was added to this function](https://github.com/scipipe/scipipe/commit/652fb4a5c367a30772db61264f3ddabe7abc1e64#diff-411fbb6fcdb986a6c7b92bb39eafecfa) back in April of this year.

So, if I go the `ip.go` file and comment out the [checkFilename() line](https://github.com/scipipe/scipipe/blob/master/ip.go#L92) then it appears to work again.

---

Anyway, I'm guessing the reason this was defined as the empty string `""` is that we don't care about the file and just want its contents? Or does that imply you want no file to be created since this is a substream?

Well, there are probably have a couple of options for solving this.

One way would be to use the built-in [temporary file feature of golang](https://golang.org/pkg/io/ioutil/#TempFile) to just create a unique temporary file somewhere; perhaps in `/tmp/`?

Another way would be to make an alternative version of [this NewFileIP function](https://github.com/scipipe/scipipe/blob/master/ip.go#L90-L105) where the default assumes the empty string:

```go
func NewSubsteamIP() *FileIP {
        ip := &FileIP{
                BaseIP:    NewBaseIP(""),
                lock:      &sync.Mutex{},
                SubStream: NewInPort("in_substream"),
        }

        if ip.Exists() {
                ip.AuditInfo() // This will populate the audit info from file
        }

        return ip
}
```

Possibly there are other ways to resolve this, depending on what you want to do. I can make a PR later if you know more details about this or have a preference.

Any thoughts?

**Comment by samuell** [comment]
Closing this, as we merged #91 by @rbisewski . Let me know if I missed something!
